### ISSUE TITLE
Rendering Layer when not nested in Source

### ISSUE THREAD

**Issue by kengreim** [GIShub4/solid-map-gl#171]
The [docs](https://gis-hub.gitbook.io/solid-map-gl/components/layer) suggest that something like this should work with a Layer component **not** nested in a Source if you already have a Source with a defined ID elsewhere

```ts
<Layer
  sourceId={SOURCE_ID}
  style={{
    type: "line",
    paint: {
      "line-color": "hsl(100, 100%, 50%)",
      "line-width": 2,
    },
  }}
/>
```

However, to get it to work with no errors from Mapbox GL JS, I have to do the following:

```ts
<Layer
  style={{
    source: SOURCE_ID,
    type: "line",
    paint: {
      "line-color": "hsl(100, 100%, 50%)",
      "line-width": 2,
    },
  }}
/>
```

It looks like the `sourceId` prop is not actually used. I suggest changing [this line](https://github.com/GIShub4/solid-map-gl/blob/72b46e81b09a0bae10f52155c2e4f9258ccdffef/src/components/Layer/index.tsx#L94) to something like below. Also would need to make a note about precedence. Or remove `sourceId` prop and make the `style` prop be the only method to specify Source Id.
```
const sourceId: string = props.sourceId || props.style?.source || useSourceId();
```

**Comment by KaiHuebner** [comment]
Thanks for finding this.
I kept both options, sometimes it's just easier to provide the sourceID instead of providing it in the style object. 
Should be fixed in the latest release (v1.11.3).

**Comment by kengreim** [comment]
Agree that having both options is good, thanks!
